### PR TITLE
Feature/159 persistent identityserver

### PR DESCRIPTION
**Pull request by wotwot563**
## Description

Changed the identity server to have a persistent data store.

## Type of change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

login on the frontend.
restart the identity server.
now login again
the api database should only have one user.

## Link to issue

Closes: #159

### CODE REVIEW

File: Data/IdentityDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        /// <value>
+        /// The user.
+        /// </value>
+        public DbSet<IdentityUser> IdentityUser { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+
+            //modelBuilder.Properties<DateTime>().Configure(c => c.HasColumnType("datetime2"));
```
Comment (Brend-Smits) [inline Data/IdentityDbContext.cs:47]: Remove the commented out code

File: IdentityServer/Quickstart/Account/ExternalController.cs [hunk 2]
```diff
@@ -208,7 +218,6 @@ public async Task<IActionResult> Callback(string code, string state, string sess
                                                                user.Username,
```
Comment (Brend-Smits) [inline IdentityServer/Quickstart/Account/ExternalController.cs:218]: This formatting looks messed up o.0

File: IdentityServer/Startup.cs [hunk 3]
```diff
@@ -77,36 +85,38 @@ public Startup(IConfiguration configuration, IWebHostEnvironment environment)
...
+                options.Events.RaiseSuccessEvents = true;
+                if(Environment.IsDevelopment())
+                {
+                    options.IssuerUri = Config.Self.IssuerUri;
+                }
+            });
+
             builder.AddInMemoryIdentityResources(IdentityConfig.GetIdentityResources());
             builder.AddInMemoryApiResources(IdentityConfig.Apis);
             builder.AddInMemoryClients(IdentityConfig.Clients(Config));
```
Comment (Brend-Smits) [inline IdentityServer/Startup.cs:118]: I'm unsure on why this is still in memory. I think for production the Identity Resources, Api Resources and the clients should also be persistent. Something along the lines of: 

```
            builder.AddOperationalStore(options =>
                options.ConfigureDbContext = builder =>
                    builder.UseSql(Configuration.GetConnectionString("DefaultConnection")));
            builder.AddConfigurationStore(options => options.ConfigureDbContext = dbContextOptionsBuilder =>     dbContextOptionsBuilder.UseSql(Configuration.GetConnectionString("DefaultConnection")));
            builder.AddAspNetIdentity<ApplicationUser>();
            builder.AddProfileService<IdentityProfileService>();
```

This is also what IdentityServer recommends for production scenarios. 
I don't think we really need in-memory for local development either seeing as we want developers to work  as close as possible to production level.

Comment (wotwot563) [reply IdentityServer/Startup.cs:118]: the reason why this was kept in memory is that there is currently no UI where you can modify these values and running queries on a production database is not something i would encourage.

Comment (Brend-Smits) [reply IdentityServer/Startup.cs:118]: I'll make a new issue for next semester to add a proper Admin UI to manage clients, Scopes and Resources properly.

**Review by RensvdLinden — APPROVED** [review]

**Pull request merged**